### PR TITLE
Js ms 01 exercises

### PR DESCRIPTION
**Pull request by JoeShep**
I've updated old and added new exercises with resources to carry us through the end of the 1st milestone. 
Once Dom's files are merged, we can re-number everything properly.

Exercise 15 still has a couple of TODO items to fill in, but if you see anything else, let me know

### CODE REVIEW

File: 01-foundations/exercises/09-streaming-io.md [hunk 1]
```diff
@@ -74,7 +74,7 @@ $ ./example.js
 This is some text without a newline$ ./example.js # Next command
 ```
 
-## streams
+### Streams
 
 You might be familiar task runner called Gulp. It uses Node.js streams to
```
Comment (C-Stein) [inline 01-foundations/exercises/09-streaming-io.md]: nope- this crew hasn't seen gulp

File: 01-foundations/exercises/10-relational-databases-ERD.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
+## Relations databases
```
Comment (C-Stein) [inline 01-foundations/exercises/10-relational-databases-ERD.md]: should this be "relational" instead?

Comment (JoeShep) [inline 01-foundations/exercises/10-relational-databases-ERD.md]: ha! yes. Good catch

File: 01-foundations/exercises/09-streaming-io.md (hunk 1 above)
Comment (JoeShep) [inline 01-foundations/exercises/09-streaming-io.md]: No task runner at all?

Comment (JoeShep) [inline 01-foundations/exercises/09-streaming-io.md]: Well, I'll just leave it for future cohorts. Will have to explain this more fully to C17

File: 01-foundations/exercises/10-relational-databases-ERD.md [hunk 3]
```diff
@@ -38,12 +40,11 @@
 > Plain and simple, think of developing a database without an ERD as building a house without a building plan. It might be doable because you think that simply laying a brick one over another is enough to build something, however the moment somebody else takes responsibility over the project there is disaster potential.
 
 #### Wait. What's an ERD?
-An *Entity-Relationship Diagram* is a visual mapping of the relationships between tables in a database.
+An *Entity-Relationship Diagram* is a visual mapping of the relationships between tables in a database. We will work through how to make an ERD in class, and complete this exercise together.
 
-Let's walk through it.  
-1. Install the [Chinook Database](https://chinookdatabase.codeplex.com/) and unzip. Save in a location you are going to remember in about 2 minutes.  
+1. Download the [musichistory.db](./assets/musichistory.db) file, and then copy it to the folder that you created for this exercise. That file **is** the database. It contains all of the tables and data.
```
Comment (C-Stein) [inline 01-foundations/exercises/10-relational-databases-ERD.md]: I can't find this file- is it real, or is this a placeholder link?

File: 01-foundations/exercises/11-sql_01-music-history.md [hunk 4]
```diff
@@ -54,23 +52,40 @@ $make
...
+* Example:  `SELECT * FROM employees WHERE name LIKE ‘b%’`
+   * LIKE means “matches a pattern” and % means “starts with” (wildcard).
 
-Download the [musichistory.db](./assets/musichistory.db) file, and then copy it to the folder that you created for this exercise. That file **is** the database. It contains all of the tables and data. 
+* Example 2: `SELECT products.id, products.name FROM order_details JOIN products ON  order_details.product_id = products.id WHERE order_details.order_id = 10250`
 
+###### Joins
+* http://blog.codinghorror.com/a-visual-explanation-of-sql-joins/
+* With a join, you must specify how the table is joined using ON. Otherwise, you will get all the combinations possible of one table row matched with another table row.
+  * **Inner Join** matches one table with another table row for row. If one row doesn’t have its pair on another table, it will be excluded. It will only show matching row pairs.
```
Comment (C-Stein) [inline 01-foundations/exercises/11-sql_01-music-history.md]: should we mention that this is the default if one doesn't specify "inner" or "outer?"

**Comment by C-Stein** [comment]
Lots of good info here! I made a few line comments, but looks really awesome overall. 👍

File: 01-foundations/exercises/09-streaming-io.md (hunk 1 above)
Comment (C-Stein) [inline 01-foundations/exercises/09-streaming-io.md]: they saw grunt, but grunt doesn't use "pipe"

File: 01-foundations/exercises/10-relational-databases-ERD.md (hunk 3 above)
Comment (JoeShep) [inline 01-foundations/exercises/10-relational-databases-ERD.md]: It's there. IF you click the link is it not doing anything?

File: 01-foundations/exercises/11-sql_01-music-history.md (hunk 4 above)
Comment (JoeShep) [inline 01-foundations/exercises/11-sql_01-music-history.md]: Will cover that live in class

**Pull request merged**